src/test/compile-fail/unboxed-closure-sugar-used-on-struct.rs

```diff
@@ -16,6 +16,7 @@ struct Bar<A> {
 
 fn foo(b: Box<Bar()>) {
     //~^ ERROR parenthesized parameters may only be used with a trait
+    //~| ERROR the type placeholder `_` is not allowed within types on item signatures
```

Comment:
This is an annoying little regression, but from what I tried "infer everything" was the least evil (in terms of redundant diagnostics), given the constraints 1) `usize()` should be a warning and not an error and 2) lifetime resolution has some issues causing ICEs if `()` in non-traits is kept "parenthesized", I hope to fix this in another PR